### PR TITLE
piece.kind added and started on bishopmoveIsBlocked

### PR DESCRIPTION
**Pull request by EllisWaterman**

### CODE REVIEW

File: projects/website/public/chess.js [hunk 1]
```diff
@@ -190,34 +186,59 @@ const rookMoveIsBlocked = (srcCol,srcRow,dstCol,dstRow) => {
...
+                    if (board[i][j] !== 0)
+                        return true
+                }
+            }
+        }
+        return false
+}
+
+const queenMoveIsBlocked = (srcCol, srcRow, dstCol, dstRow) => {
+    if((rookMoveIsBlocked(srcCol, srcRow, dstCol, dstRow)) || (bishopMoveIsBlocked(srcCol, srcRow, dstCol, dstRow))) {
```
Comment (gigamonkey) [inline projects/website/public/chess.js:227]: Nice.

File: projects/website/public/chess.js [hunk 2]
```diff
@@ -190,34 +186,59 @@ const rookMoveIsBlocked = (srcCol,srcRow,dstCol,dstRow) => {
...
             }
+        } else if ((srcCol < dstCol) && (srcRow < dstRow)) {
+            for (let i = srcCol + 1; i < dstCol; i++) {
+                for (let j = srcRow + 1; j < dstRow; j++) {
+                    console.log(board[i][j])
+                    if (board[i][j] !== 0)
+                        return true
+                }
+            }
+        }
```
Comment (gigamonkey) [inline projects/website/public/chess.js:222]: I'm too tired to actually work it all out but I'm pretty sure you could simplify this by figuring out two values: `rowChange` and `colChange` which would each be either -1 or +1 and then write a single loop that changes the row by and column by those two values to traverse from `srcRow` and `srcCol` to `dstRow` and `dstCol`.

**Review by gigamonkey — APPROVED** [review]
Really great work on this project!

File: projects/website/public/chess.js (hunk 1 above)
Comment (EllisWaterman) [reply projects/website/public/chess.js:227]: this does not work

Comment (EllisWaterman) [reply projects/website/public/chess.js:227]: the queen move is blocked

File: projects/website/public/chess.js [hunk 3]
```diff
@@ -1,332 +1,368 @@
...
-                }
-            }
-        }
-        return false
-}
-
-const queenMoveIsBlocked = (srcCol, srcRow, dstCol, dstRow) => {
-    if((rookMoveIsBlocked(srcCol, srcRow, dstCol, dstRow)) || (bishopMoveIsBlocked(srcCol, srcRow, dstCol, dstRow))) {
-        return true
+  if (srcCol < dstCol && srcRow > dstRow) {
```
Comment (gigamonkey) [inline projects/website/public/chess.js:201]: You have four very similar loops in this function, that each figures out how many squares to check and then moves `col` and `row` by a certain amount. I bet you can collapse these down to one loop plus maybe a few extra variables to capture the differences between the loops. Ask me about it in class if you don't see how.

**Review by gigamonkey — APPROVED** [review]
This seems like good progress. Go ahead and merge it because I think we did a big reformat which is making the diff quite large. But keep the branch for future work.

**Pull request merged**